### PR TITLE
MB-748: ma - customer can log in

### PR DESCRIPTION
**Pull request by rswerve**
## Description

This is the story for adding roles to users.  3 points seems to have been a slight underestimate :-), so opening a draft PR to get folks looking at the changes.

## Reviewer Notes

Success here is if there is a populated roles object in userInfo in Redux.  See the screenshot below.

In order to test:
`make db_dev_migrate`
`make db_dev_e2e_populate`
`make server_run`
`make client_run`
Local sign in
log in as `role_tester@service.mil`
Check the Redux store in dev tools.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-748) for this change

## Screenshots

image

### CODE REVIEW

File: pkg/models/roles.go [hunk 1]
```diff
@@ -1,28 +1,23 @@
...
 )
 
 // Role is an object representing the types of users who can authenticate in the admin app
-type Role struct {
-	ID        int       `json:"id" db:"id"`
-	RoleType  string    `json:"role_type" db:"role_type"`
-	CreatedAt time.Time `json:"created_at" db:"created_at"`
-	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
-}
+type Role auth.Role
```
Comment (mkrump) [inline pkg/models/roles.go]: This seems a little weird having some of the `Role` model information here in the models package (i.e. Validate) and the rest in `auth`. Any reason why it's not all in one place?

File: migrations/20191210193133_update_roles_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
+ALTER TABLE user_roles RENAME TO users_roles;
```
Comment (mkrump) [inline migrations/20191210193133_update_roles_table.up.sql]: I don't think these were added here but when i looked at `roles` there was a duplicate for `customer`.
Screen Shot 2019-12-16 at 9 42 10 AM

Comment (rswerve) [reply migrations/20191210193133_update_roles_table.up.sql]: Added a constraint for this.

File: pkg/models/roles.go (hunk 1 above)
Comment (rswerve) [reply pkg/models/roles.go]: It's weird.  We punted for a future story, because it looks to be fairly involved to fix.

Comment (mkrump) [reply pkg/models/roles.go]: 👍

**Pull request merged**